From stable-bounces@linux.kernel.org Wed Oct  4 02:17:45 2006
Message-Id: <200610040915.k949FZY8031823@shell0.pdx.osdl.net>
To: torvalds@osdl.org
From: Jim Cromie <jim.cromie@gmail.com>
Date: Wed, 04 Oct 2006 02:15:35 -0700
Cc: akpm@osdl.org, jim.cromie@gmail.com, johnstul@us.ibm.com,
        alexander.krause@erazor-zone.de, dzpost@dedekind.net,
        phelps@mantara.com, stable@kernel.org
Subject: scx200_hrt: fix precedence bug manifesting as 27x clock in 1 MHz mode

From: Jim Cromie <jim.cromie@gmail.com>

Fix paren-placement / precedence bug breaking initialization for 1 MHz
clock mode.

Also fix comment spelling error, and fence-post (off-by-one) error on
symbol used in request_region.

Addresses http://bugzilla.kernel.org/show_bug.cgi?id=7242

Thanks alexander.krause@erazor-zone.de, dzpost@dedekind.net, for the
reports and patch test, and phelps@mantara.com for the independent patch
and verification.

Signed-off-by:  Jim Cromie <jim.cromie@gmail.com>
Cc: <alexander.krause@erazor-zone.de>
Cc: <dzpost@dedekind.net>
Cc: <phelps@mantara.com>
Acked-by: John Stultz <johnstul@us.ibm.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/clocksource/scx200_hrt.c |    4 ++--
 include/linux/scx200.h           |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- linux-2.6.18.orig/drivers/clocksource/scx200_hrt.c
+++ linux-2.6.18/drivers/clocksource/scx200_hrt.c
@@ -63,7 +63,7 @@ static struct clocksource cs_hrt = {
 
 static int __init init_hrt_clocksource(void)
 {
-	/* Make sure scx200 has initializedd the configuration block */
+	/* Make sure scx200 has initialized the configuration block */
 	if (!scx200_cb_present())
 		return -ENODEV;
 
@@ -76,7 +76,7 @@ static int __init init_hrt_clocksource(v
 	}
 
 	/* write timer config */
-	outb(HR_TMEN | (mhz27) ? HR_TMCLKSEL : 0,
+	outb(HR_TMEN | (mhz27 ? HR_TMCLKSEL : 0),
 	     scx200_cb_base + SCx200_TMCNFG_OFFSET);
 
 	if (mhz27) {
--- linux-2.6.18.orig/include/linux/scx200.h
+++ linux-2.6.18/include/linux/scx200.h
@@ -32,7 +32,7 @@ extern unsigned scx200_cb_base;
 
 /* High Resolution Timer */
 #define SCx200_TIMER_OFFSET 0x08
-#define SCx200_TIMER_SIZE 0x05
+#define SCx200_TIMER_SIZE 0x06
 
 /* Clock Generators */
 #define SCx200_CLOCKGEN_OFFSET 0x10
